client/config.go

```diff
@@ -148,32 +150,39 @@ func (c Configuration) NewAdminClient(
 	writeRequestScope := iopts.MetricsScope().SubScope("write-req")
 	fetchRequestScope := iopts.MetricsScope().SubScope("fetch-req")
 
-	topoInit := params.TopologyInitializer
-	if topoInit == nil {
-		if c.ConfigService == nil {
-			return nil, errConfigurationMustSupplyConfigService
-		}
+	configureResults := &environment.ConfigureResults{
+		TopologyInitializer: params.TopologyInitializer,
+	}
 
-		configSvcClient, err := c.ConfigService.NewClient(iopts)
-		if err != nil {
-			return nil, err
+	if configureResults.TopologyInitializer == nil {
+		switch {
+		case c.EnvironmentConfig.Service != nil:
+			configSvcClient, err := c.EnvironmentConfig.Service.NewClient(iopts)
+			if err != nil {
+				return nil, err
+			}
+
+			configureResults, err = c.EnvironmentConfig.NewConfigureResults(configSvcClient, iopts, "", c.HashingConfiguration.Seed)
+			if err != nil {
+				err = fmt.Errorf("unable to create dynamic topology initializer, err: %v", err)
+				return nil, err
+			}
+
+		case c.EnvironmentConfig.Static != nil:
+			var err error
+			configureResults, err = c.EnvironmentConfig.NewConfigureResults(nil, nil, c.ListenAddress, 0)
```

Comment:
Yeah let's move `ListenAddress` into `environment.StaticConfiguration` as it's pretty strange to specify it here.